src/uri/authority.rs

```diff
@@ -23,15 +23,9 @@ impl Authority {
 
     // Not public while `bytes` is unstable.
     pub(super) fn from_shared(s: Bytes) -> Result<Self, InvalidUri> {
-        let authority_end = Authority::parse_non_empty(&s[..])?;
-
-        if authority_end != s.len() {
-            return Err(ErrorKind::InvalidUriChar.into());
-        }
-
-        Ok(Authority {
-            data: unsafe { ByteStr::from_utf8_unchecked(s) },
-        })
+        // Preconditon on create_authority: trivially satisfied by the
```

Comment:
typo:
```suggestion
        // Precondition on create_authority: trivially satisfied by the
```

Reply:
Thank you for this. I have applied your suggestion.

Reply:
Thank you for this. I have applied your correction.